_rules/SC4-1-2-role-attribute-has-valid-value.md

```diff
@@ -0,0 +1,123 @@
+---
+name: Role attribute has valid value
+test_type: atomic
+
+description: |
+   This rule checks that role attributes have valid values
+   
+success_criterion:
+- 4.1.2 # Name, Role, Value
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Jey Nandakumar
+---
+
+## Test procedure
+
+### Applicability
+
+Any [non-empty](#non-empty) `role` attribute that is placed on an HTML or SVG element that is [exposed to assistive technologies](#exposed-to-assistive-technologies).
+
+### Expectation
+
+Each test target has at least one non-abstract valid value as specified in the [WAI-ARIA 1.1, roles](https://www.w3.org/TR/wai-aria-1.1/#roles_categorization).
+
+## Assumptions
+
+*There are currently no assumptions*
+
+## Accessibility Support
+
+Older browsers do not support fallback roles. If multiple values are used in the role attribute, the attribute is ignored.
+
+At the moment, none of the major assistive technologies support all roles.
```

Comment:
Suggest making "roles" a bit more explicit, e.g.:
"At the moment, none of the major assistive technologies support all `[WAI-ARIA 1.1 roles](https://www.w3.org/TR/wai-aria-1.1/#roles_categorization).`"